Added native binding for dictionary duplication

2shady4u opened the pull request:
There was no native binding for Dictionary duplication.
I took the native binding for array duplication (from modules/gdnative/gdnative/array.cpp)
and modified it for dictionaries.

Was this oversight deliberate? or didn't anyone need to duplicate their dictionaries in native scripts until now?

karroffel reviewed (changes requested):
This seems fine but the function won't be able to be called from any bindings without an entry in the `gdnative_api.json`

2shady4u commented:
@karroffel I have implemented the requested changes and added entries for the duplicate function in both `gdnative_api.json` and `dictionary.h`

Should hopefully be acceptable now.... Unless I've forgotten something else?

karroffel commented on modules/gdnative/gdnative_api.json:
```diff
@@ -3122,6 +3122,14 @@
           ["godot_dictionary *", "p_self"]
         ]
       },
+      {
```
That's almost correct. Once we made a release we must not change the old API entries. Instead a "core extension" has to be made for this. There already is the 1.1 extension, this would have to go into the 1.2 extension.

karroffel reviewed (changes requested):
Ups, I meant this. :sweat_smile:

2shady4u committed:
Added native binding for dictionary duplication


Added entry in gdnative_api.json


Added function to header as well


Fixed versioning

karroffel reviewed (approved):
Seems good now :)

akien-mga commented:
Thanks!
